lib/AST/PluginLoader.cpp

```diff
@@ -106,8 +106,19 @@ PluginLoader::getPluginMap() {
     case PluginSearchOption::Kind::LoadPluginExecutable: {
       auto &val = entry.get<PluginSearchOption::LoadPluginExecutable>();
       assert(!val.ExecutablePath.empty() && "empty plugin path");
-      for (auto &moduleName : val.ModuleNames) {
-        try_emplace(moduleName, /*libraryPath=*/"", val.ExecutablePath);
+      if (llvm::sys::path::filename(val.ExecutablePath).ends_with(".wasm")) {
+        // we treat wasm plugins like library plugins that can be loaded by an external
+        // "macro runner" executable
+        SmallString<128> runner;
+        // TODO: improve path resolution: we really want tools_dir
+        llvm::sys::path::append(runner, Ctx.SearchPathOpts.RuntimeResourcePath, "../../bin/swift-wasm-plugin-server");
```

Comment:
This seems like it's not the best way to find the binary — is there a better way to locate it from the frontend or should we hand it down from the driver?

Reply:
Decided to tentatively use the Driver approach.